MonoGame.Framework/Utilities/Png/PngCommon.cs

```diff
@@ -0,0 +1,685 @@
+﻿// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections;
+using System.Text;
+using Microsoft.Xna.Framework;
+
+namespace MonoGame.Utilities.Png
+{
+    public class Palette
```

Comment:
Can you go thru and make all these internal objects `internal` and not `public`?